src/main.cpp

```diff
@@ -7028,9 +7034,12 @@ bool SendMessages(CNode *pto)
 {
     const Consensus::Params &consensusParams = Params().GetConsensus();
     {
-        // Don't send anything until we get its version message otherwise we may
-        // end up geting ourselves banned by the receiving peer.
-        if (pto->nVersion == 0)
+        // First set fDisconnect if appropriate.
+        pto->DisconnectIfBanned();
+
+        // Now exit early if disconnecting or the version handshake is not complete.  We must not send PING or other
+        // connection maintenance messages before the handshake is done.
+        if (pto->fDisconnect || !pto->fSuccessfullyConnected)
             return true;
```

Comment:
I don't see how it's possible for us to send a VERACK  if we can only send messages when successfully connected (which requires that we received a VERACK)?  so how will we ever receive a VERACK or be able to send them from a BU node?   I'll run the code anyway to verify, maybe I"m not reading this correctly...